beacon_node/lighthouse_network/src/rpc/handler.rs

```diff
@@ -967,7 +967,10 @@ where
                     OutboundRequestContainer {
                         req: req.clone(),
                         fork_context: self.fork_context.clone(),
-                        max_rpc_size: max_rpc_size(&self.fork_context),
+                        max_rpc_size: max_rpc_size(
+                            &self.fork_context,
+                            self.listen_protocol.upgrade().max_rpc_size,
+                        ),
```

Comment:
```suggestion
                        max_rpc_size: self.listen_protocol.upgrade().max_rpc_size,
```
what we pass here should be the chunk size, the `rpc_max_size` calculated for the upgrade has already done this.

However this to me mean means there might a bug in unstable, since the `SubstreamProtocol`  (what `upgrade()` returns) has both the `max_rpc_size` and `fork_context`. `max_rpc_size` is calculated with the fork present at the moment the upgrade params are set (first time the handler is created) so in theory it shouldn't manage 100% correctly fork upgrades as it's working right now. 

`SubstreamProtocol` should store the `rpc_chunk_size` instead of `rpc_max_size` and calculate the `rpc_max_size` on the fly each time